aiida/tools/dbexporters/tcod.py

```diff
@@ -268,16 +267,27 @@ def decode_textfield_ncr(content):
     """
     Decodes the contents for CIF textfield from Numeric Character Reference.
 
-    :param content: a string with contents
-    :return: decoded string
+    :param content: contents as bytes
+    :return: decoded string as bytes
     """
     import re
 
     def match2str(m):
-        if isinstance(m.group(1), str):
-            return chr(int(m.group(1)))
+        """
+        Function returns a byte with a value of the first group of regular
+        expression. Different methods to convert int -> byte are used for
+        Python 2 and Python 3 as there seems to be no common way to do so.
+        chr() in Python 3 returns value in Unicode format which is incompatible
+        with the requested return value of bytes format.
+
+        :param match: match result of re.sub
+        :return: a single byte having a value of the first group in re.sub
+        """
+        byte_value = int(m.group(1))
+        if six.PY2:
+            return chr(byte_value)
         else:
-            return bytes([int(m.group(1))])
+            return bytes([byte_value])
```

Comment:
Just realized that six has the int2byte function:
https://pythonhosted.org/six/#six.int2byte
Better to use this

Reply:
Fixed, thanks!